src/mod/dns.mod/coredns.c

```diff
@@ -1196,6 +1196,42 @@ static void dns_lookup(sockname_t *addr)
   sendrequest(rp, T_PTR);
 }
 
+static int dns_hosts(char *hostn) {
+  FILE* f;
+  char line[1024];
+  char *ptr;
+  int l = strlen(hostn);
+  const char *sep = " \t\n\v\f\r";
+  sockname_t name;
+
+  f = fopen("/etc/hosts", "rb");
+  if (!f) {
+    ddebug1(RES_MSG "fopen(): %s ", strerror(errno));
+    return 0;
+  }
+  while (fgets(line, sizeof line , f)) {
+    if ((ptr = strchr(line, '#')))
+      *ptr = '\0';
+    for(ptr = strstr(line, hostn); ptr; ptr = strstr(ptr + l, hostn)) {
+      if ((isspace(ptr[l]) || !ptr[l]) && ptr != line) {
```

Comment:
Is `ptr != line` intended to exclude it as the first word? then this would need space-skipping too. Maybe space-skip "line" as a whole before the loop?

Reply:
Yes and yes.

Reply:
btw, strstr(ptr + l) looks like a bug.